docs/proposals/33-machine-image-capabilities.md

```diff
@@ -188,22 +188,19 @@ In this case it does not result in an error but in performance loss.
 
 ## Proposal
 
-Introduce a top level capabilities map in the CloudProfile `spec.capabilities`.
-
-Capabilities are very specific to the provider and the selected catalog offered by Gardener in the cloud profile.
-To minimize complexity and data size in the cloud profile, the capabilities are defined as a map with string keys and string arrays as values.
-The key is the capability name and the value is an array of possible values.
+Introduce a top level capabilities array in the CloudProfile `spec.capabilities`.
 
 ```go
 type Spec struct {
-    Capabilities map[string][]string `json:"capabilities"`
+    Capabilities []map[string][]string
```

Comment:
Is this structure even possible or will we run into Protobuf issues (again)?

Frankly, I don't like this data structure much as it's rather unhandy/confusing for implementations. WDYT about the following proposal instead?

```go
Capabilities []Capability

type Capability struct {
    Name string
    Values []string
}
```

cc @LucaBernstein

Reply:
would also be fine for me.